pkg/keycloak/client.go

```diff
@@ -678,25 +403,27 @@ func defaultRequester() Requester {
 }
 
 type KeycloakInterface interface {
-	ListRealms() ([]*v1alpha1.KeycloakRealm, error)
-	GetRealm(realmName string) (*v1alpha1.KeycloakRealm, error)
 	CreateRealm(realm *v1alpha1.KeycloakRealm) error
-	UpdateRealm(realm *v1alpha1.KeycloakRealm) error
+	GetRealm(realmName string) (*v1alpha1.KeycloakRealm, error)
+	UpdateRealm(kcRealm, objRealm *v1alpha1.KeycloakRealm) error
```

Comment:
I think the signature of these update functions are wrong. We removed the need for the `kc` object by adding any necessary fields to the `spec` object (like the ID) before calling the update function